github/compliance/milestone.py

```diff
@@ -3,26 +3,51 @@
 import delivery.model
 
 
-def _milestone_title(sprint: delivery.model.Sprint):
+def _milestone_title(sprint: delivery.model.Sprint) -> str:
     return f'sprint-{sprint.name}'
 
 
 def find_sprint_milestone(
     repo: github3.repos.Repository,
-    sprint: delivery.model.Sprint,
-):
-    for ms in repo.milestones():
-        if ms.title == _milestone_title(sprint=sprint):
-            return ms
-    return None
+    sprints: tuple[delivery.model.Sprint],
+) -> tuple[delivery.model.Sprint, github3.repos.repo.milestone.Milestone]:
+    sprint_milestone = None
+    all_milestones = repo.milestones(state='all')
```

Comment:
does `repo.milestones()` perform an HTTP request?
if so, we could consider some caching (or passing in as for sprints) to reduce HTTP load

Reply:
from how github3.py behaves elsewhere, I think this should return an iterator that will issue an http-request upon consuming results. rather than caching (which would be problematic in this case, I think), we might consider limiting the amount of milestones + order by date. for the time being, we have some few tens of milestones (rate of growth: ~30-ish / year). -> nothing to worry about short-term..